### ISSUE TITLE
API work: add committee filing frequency to /committees/ endpoint

### ISSUE THREAD

**Issue by PaulClark2** [fecgov/openFEC#3049]
**What we’re after**:
So that users can filter out terminated and administratively terminated committees, we should add committee filing frequency to /committees/ endpoint.

**Current state**:
Users cannot exclude terminated and administratively terminated committees. The web app displays terminated and administratively terminated committees with no indication they might be terminated. 
https://www.fec.gov/data/committees/?committee_type=V&committee_type=W

**Completion criteria**:
- [ ] add the field to the schema, which is what outputs the vaule
- [ ] add a filter _multi on the value (terminated and administratively terminated committees--A, T, M or Q are options
- [ ] add an index to MV for filing frequency (to both the committee detail and committee history materialized views)

**Comment by rjayasekera** [comment]
I have analyzed this issue furter and here are my findings.

https://www.fec.gov/data/committees/?q=C00000018 page uses 
https://api.open.fec.gov/v1/committees/?committee_id=C00000018&api_key=DEMO_KEY end point
and it queries public.ofec_committee_detail_mv which contains the last filing of unique committees

So if the committee filing_frequency is set for "T" or "A" at the last filing this is what end up in the committee_details table.

```
select committee_id,cycle,filing_frequency,cycles,* from public.ofec_committee_history_mv 
where committee_id='C00000018' order by committee_id, cycle limit 20;
+----------------+---------+--------------------+--------------------------------------------+--------+---------+---------
| committee_id   | cycle   | filing_frequency   | cycles                                     | idx    | cycle   | committe
|----------------+---------+--------------------+--------------------------------------------+--------+---------+---------
| C00000018      | 1976    | Q                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 948966 | 1976    | C0000001
| C00000018      | 1978    | Q                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 91791  | 1978    | C0000001
| C00000018      | 1980    | Q                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 157944 | 1980    | C0000001
| C00000018      | 1982    | Q                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 538388 | 1982    | C0000001
| C00000018      | 1984    | Q                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 538387 | 1984    | C0000001
| C00000018      | 1986    | Q                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 776431 | 1986    | C0000001
| C00000018      | 1988    | T                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 776430 | 1988    | C0000001
+----------------+---------+--------------------+--------------------------------------------+--------+---------+---------

select committee_id,cycle,filing_frequency,cycles,* from public.ofec_committee_detail_mv 
where committee_id='C00000018' order by committee_id, cycle limit 20;
+----------------+---------+--------------------+--------------------------------------------+--------+---------+---------
| committee_id   | cycle   | filing_frequency   | cycles                                     | idx    | cycle   | committe
|----------------+---------+--------------------+--------------------------------------------+--------+---------+---------
| C00000018      | 1988    | T                  | [1986, 1984, 1976, 1982, 1980, 1978, 1988] | 776430 | 1988    | C0000001
+----------------+---------+--------------------+--------------------------------------------+--------+---------+---------
```
I have expose filing_frequency field in the API endpoint and it can be used in queries.  Just wanted point out that value of filing_frequency in committee_details is for the committees filing status of the last filing and it does not corresponding to any election previous cycles.

**Comment by ccostino** [comment]
There's been a bit of confusion here in terms of what's desired; based on what I'm reading in the description my understanding is this:

* Expose the `filing_frequency_type` field in the API output (and add indexes to the materialized views to support it where appropriate)
* Provide a way to filter out any results of `filing_frequency_type` that match `T` or `A` (could be through selecting one or more types, or a simple boolean check that displays/doesn't display them in the results on the frontend)
* Make these changes to the `/committees` endpoint

@rjayasekera brings up a good point about the fact that the `/committees` endpoint operates off of the `ofec_committee_details_mv` materialized view, which as he explained only shows the final record from the committee history.  The work that's already been so far does expose the `filing_frequency_type` in the committee history as well (which operates off of the `ofec_committee_history_mv` materialized view), but that endpoint is specific to a committee; you can't just browse the whole thing.

Is our understanding correct, @PaulClark2?  Is there something else desired here instead?  If you could please clarify either in the details above or just comment in response, we'd appreciate it. :-)  Thanks!

**Comment by PaulClark2** [comment]
@ccostino @rjayasekera this issue was entered because an external user asked what the correct number of hybrid (Carey) PACs. https://www.fec.gov/data/committees/?cycle=2018&committee_type=V&committee_type=W (The webapp does not apply the committee type selections contained in the URL. That's a different problem.) I added `?cycle=2018`. (Another separate issue, the webapp doesn't help the user understand what data/time frame they are looking at.)

The webapp returns 272 hybrid committees. This response is misleading. Only 199 of them are active. The other 73 are terminated or administratively terminated. I think our default display should show folks active committees and the filter panel gives them the option to include terminated and administratively terminated committees.

We should be able to display the correct status of a committee (actively filing (monthly, quarterly) or terminated (terminated, administratively terminated) for the time period selected by a user. 

`select cmte_id, cmte_tp, cmte_filing_freq from DISCLOSURE.CMTE_VALID_FEC_YR where fec_election_yr = 2018 and cmte_tp in ('V', 'W') order by 3;`
272

`select * from DISCLOSURE.CMTE_VALID_FEC_YR where fec_election_yr = 2018 and cmte_tp in ('V', 'W') and cmte_filing_freq not in ('T', 'A');`
199

2017-18 hybrid PACs

CMTE_ID	|	CMTE_TP	|	CMTE_FILING_FREQ
----	|	----	|	----
C00489658	|	V	|	A
C00496927	|	V	|	A
C00500009	|	V	|	A
C00501262	|	V	|	A
C00529743	|	V	|	A
C00540138	|	V	|	A
C00542969	|	V	|	A
C00545996	|	V	|	A
C00561092	|	V	|	A
C00566018	|	V	|	A
C00572529	|	V	|	A
C00572776	|	V	|	A
C00575761	|	V	|	A
C00576132	|	V	|	A
C00576892	|	V	|	A
C00577015	|	V	|	A
C00577064	|	V	|	A
C00578286	|	V	|	A
C00581942	|	V	|	A
C00583419	|	V	|	A
C00583450	|	V	|	A
C00584839	|	V	|	A
C00585398	|	V	|	A
C00589176	|	V	|	A
C00590869	|	V	|	A
C00594523	|	V	|	A
C00600684	|	V	|	A
C00606996	|	V	|	A
C00613844	|	V	|	A
C00614339	|	V	|	A
C00614693	|	V	|	A
C00618462	|	V	|	A
C00618496	|	V	|	A
C00618652	|	V	|	A
C00620096	|	V	|	A
C00622787	|	V	|	A
C00623249	|	V	|	A
C00625061	|	V	|	A
C00626267	|	V	|	A
C00626424	|	V	|	A
C00626705	|	V	|	A
C00627711	|	V	|	A
C00628727	|	V	|	A
C00628834	|	V	|	A
C00630111	|	V	|	A
C00630368	|	V	|	A
C00630624	|	V	|	A
C00633420	|	V	|	A
C00383182	|	V	|	T
C00661660	|	V	|	T
C00496505	|	W	|	T
C00506469	|	V	|	T
C00509489	|	V	|	T
C00550400	|	V	|	T
C00564773	|	V	|	T
C00583617	|	V	|	T
C00605162	|	V	|	T
C00608166	|	W	|	T
C00612101	|	V	|	T
C00619577	|	V	|	T
C00623538	|	V	|	T
C00623769	|	V	|	T
C00623967	|	V	|	T
C00629352	|	V	|	T
C00630814	|	V	|	T
C00630822	|	V	|	T
C00632224	|	V	|	T
C00633180	|	V	|	T
C00633412	|	V	|	T
C00633768	|	V	|	T
C00640177	|	V	|	T
C00661249	|	V	|	T
C00476838	|	V	|	T
C00001727	|	W	|	M
C00250399	|	W	|	M
C00298000	|	W	|	M
C00327189	|	W	|	M
C00342048	|	W	|	M
C00401224	|	W	|	M
C00402800	|	V	|	M
C00410068	|	W	|	M
C00428557	|	V	|	M
C00455378	|	W	|	M
C00488478	|	V	|	M
C00495861	|	V	|	M
C00540443	|	W	|	M
C00544767	|	W	|	M
C00559740	|	W	|	M
C00562777	|	W	|	M
C00608489	|	W	|	M
C00616250	|	V	|	M
C00632133	|	V	|	M
C00632562	|	V	|	M
C00633248	|	W	|	M
C00641555	|	V	|	M
C00650754	|	V	|	M
C00650861	|	V	|	M
C00651729	|	V	|	M
C00653345	|	V	|	M
C00653352	|	V	|	M
C00654467	|	V	|	M
C00661488	|	V	|	M
C00010322	|	W	|	Q
C00159319	|	W	|	Q
C00229385	|	W	|	Q
C00300921	|	W	|	Q
C00324871	|	W	|	Q
C00341396	|	W	|	Q
C00359992	|	W	|	Q
C00418897	|	W	|	Q
C00427435	|	W	|	Q
C00433227	|	V	|	Q
C00437376	|	W	|	Q
C00452276	|	W	|	Q
C00461772	|	W	|	Q
C00463331	|	W	|	Q
C00468447	|	W	|	Q
C00468868	|	V	|	Q
C00475582	|	W	|	Q
C00476978	|	W	|	Q
C00492124	|	V	|	Q
C00493510	|	V	|	Q
C00503417	|	V	|	Q
C00507053	|	W	|	Q
C00512608	|	V	|	Q
C00514653	|	V	|	Q
C00515064	|	V	|	Q
C00516500	|	W	|	Q
C00519413	|	W	|	Q
C00520080	|	V	|	Q
C00524280	|	W	|	Q
C00524355	|	V	|	Q
C00525899	|	V	|	Q
C00528307	|	W	|	Q
C00528968	|	V	|	Q
C00540229	|	V	|	Q
C00540997	|	V	|	Q
C00542217	|	V	|	Q
C00542258	|	V	|	Q
C00545004	|	V	|	Q
C00545897	|	V	|	Q
C00546481	|	V	|	Q
C00547117	|	V	|	Q
C00553313	|	W	|	Q
C00557025	|	V	|	Q
C00564690	|	V	|	Q
C00567560	|	V	|	Q
C00569962	|	V	|	Q
C00570416	|	V	|	Q
C00570903	|	V	|	Q
C00572537	|	V	|	Q
C00573642	|	V	|	Q
C00573790	|	V	|	Q
C00574061	|	V	|	Q
C00577445	|	V	|	Q
C00577759	|	V	|	Q
C00578187	|	V	|	Q
C00578997	|	V	|	Q
C00580993	|	V	|	Q
C00584482	|	V	|	Q
C00585562	|	V	|	Q
C00587030	|	V	|	Q
C00587717	|	V	|	Q
C00589507	|	V	|	Q
C00599050	|	V	|	Q
C00600817	|	V	|	Q
C00602797	|	V	|	Q
C00602813	|	V	|	Q
C00603084	|	V	|	Q
C00609396	|	V	|	Q
C00609677	|	V	|	Q
C00610568	|	W	|	Q
C00611517	|	V	|	Q
C00617738	|	V	|	Q
C00620427	|	W	|	Q
C00625145	|	W	|	Q
C00625343	|	V	|	Q
C00625731	|	V	|	Q
C00626671	|	V	|	Q
C00629006	|	V	|	Q
C00629147	|	V	|	Q
C00630103	|	V	|	Q
C00630459	|	V	|	Q
C00630699	|	V	|	Q
C00630830	|	V	|	Q
C00630921	|	V	|	Q
C00631069	|	V	|	Q
C00631481	|	V	|	Q
C00631804	|	V	|	Q
C00632554	|	V	|	Q
C00632661	|	V	|	Q
C00632786	|	V	|	Q
C00633404	|	V	|	Q
C00634279	|	V	|	Q
C00634394	|	V	|	Q
C00634501	|	V	|	Q
C00634675	|	V	|	Q
C00634964	|	V	|	Q
C00635391	|	V	|	Q
C00635649	|	V	|	Q
C00635664	|	V	|	Q
C00635748	|	V	|	Q
C00636027	|	V	|	Q
C00636639	|	V	|	Q
C00637736	|	V	|	Q
C00638320	|	V	|	Q
C00638817	|	V	|	Q
C00638866	|	V	|	Q
C00638916	|	V	|	Q
C00639922	|	V	|	Q
C00640185	|	V	|	Q
C00640367	|	V	|	Q
C00646364	|	V	|	Q
C00646919	|	V	|	Q
C00647784	|	V	|	Q
C00649053	|	V	|	Q
C00651349	|	V	|	Q
C00652859	|	V	|	Q
C00653311	|	V	|	Q
C00654210	|	V	|	Q
C00654269	|	V	|	Q
C00654814	|	V	|	Q
C00655241	|	V	|	Q
C00655936	|	V	|	Q
C00655969	|	V	|	Q
C00656330	|	V	|	Q
C00656413	|	V	|	Q
C00656660	|	V	|	Q
C00658039	|	V	|	Q
C00658211	|	V	|	Q
C00658302	|	V	|	Q
C00658971	|	V	|	Q
C00659102	|	V	|	Q
C00659805	|	V	|	Q
C00660290	|	V	|	Q
C00660589	|	V	|	Q
C00660985	|	V	|	Q
C00661041	|	V	|	Q
C00662148	|	V	|	Q
C00662254	|	V	|	Q
C00662312	|	V	|	Q
C00662387	|	V	|	Q
C00662890	|	V	|	Q
C00663377	|	V	|	Q
C00663385	|	V	|	Q
C00663393	|	V	|	Q
C00663963	|	V	|	Q
C00664433	|	V	|	Q
C00665240	|	V	|	Q
C00666123	|	V	|	Q
C00666727	|	V	|	Q
C00667055	|	V	|	Q
C00668038	|	V	|	Q
C00668178	|	V	|	Q
C00668202	|	V	|	Q
C00669259	|	V	|	Q
C00670216	|	V	|	Q
C00670554	|	V	|	Q
C00670596	|	V	|	Q
C00671040	|	V	|	Q
C00672006	|	V	|	Q
C00672576	|	V	|	Q
C00674093	|	V	|	Q
C00674572	|	V	|	Q
C00674655	|	V	|	Q
C00674697	|	V	|	Q
C00675132	|	V	|	Q
C00675314	|	V	|	Q
C00675322	|	V	|	Q
C00676270	|	V	|	Q
C00676304	|	V	|	Q
C00677161	|	V	|	Q

**Comment by PaulClark2** [comment]
@ccostino @rjayasekera  yes to these two if we are displaying filing frequency information for the period selected by the user.

- Expose the filing_frequency_type field in the API output (and add indexes to the materialized views to support it where appropriate)
- Provide a way to filter out any results of filing_frequency_type that match T or A (could be through selecting one or more types, or a simple boolean check that displays/doesn't display them in the results on the frontend)

Probably yes to this one, too. 

- Make these changes to the /committees endpoint

**Comment by ccostino** [comment]
Thank you, @PaulClark2!

For the very first point about what the web app is doing (or not doing in this case) with the filters in the URL, yeah, that is an issue. :-(  If we don't already have something filed for it we should; I took a cursory look through the CMS repo but didn't see one specific to this behavior.

I did find the issue for [designing this interaction](https://github.com/fecgov/fec-cms/issues/1944) that these API changes would support, though, and based on the design request and what you outlined above I think this makes sense but we might still be talking about two different things. :-)

First, the main issue at hand:  What's being asked for is that by default, we should not show terminated or administratively terminated committees in any of the results we're browsing unless the user explicitly requests to.  This is regardless of whatever other filters are applied, including whether or not the user selected a specific cycle/cycles.  Is this statement correct in capturing the desired behavior?

If it is, what @rjayasekera pointed out to me is that if you select a committee and filter exclusively for a cycle(s) that the committee was active and nothing else, the committee would still not show up in the results because it'd be hidden by default due to its `filing_frequency_type` being `T` or `A`.  This is consistent with the behavior being described above, but may not be the end result we're going for and that's what we're trying to double check.

The `ofec_committee_details_mv` materialized view that powers the the `/committees` endpoint only contains a single record for a given committee, which is derived from the last record entered for a committee in the `ofec_committee_history_mv` materialized view.  The record contains a list of all cycles the committee was operational/active, which is why we're able to filter by cycle on the endpoint, but the `filing_frequency_type` is only going to match the most recent value from those records in the committee history.  If we should instead be showing the committee in this scenario, where a user selects a cycle(s) where the committee was not terminated yet, we'll have to go back to the drawing board a bit to work out how that would be possible.

It'd probably help to just do a live demonstration of this with the work @rjayasekera has done so far and talk through it some more together.  As it stands, the work so far will meet the base criteria by providing a way for a user to filter out or exclude the `filing_frequency_type` values of `T` or `A`, but we just want to be sure. :-)

**Comment by rjayasekera** [comment]
I am 95% done of this issue, but did not commit any changes to sprint 5.5.  I think no need to rush this issue into Sprint 5.5, since we don't have the front end implementation ready for it yet.  I will continue to work on it and  complete this in Sprint 5.6

**Comment by lbeaufort** [comment]
@rjayasekera great work! Do you mind putting up a [WIP] PR with what you have so far?

**Comment by ccostino** [comment]
@rjayasekera, for future reference this kind of work is okay to get in place and released before any front-end work is done.  This adjustment isn't impacting any existing endpoints as it is just exposing a bit of extra information, so those changes are safe to go out beforehand. :-)

**Comment by JonellaCulmer** [comment]
@PaulClark2 Is issue https://github.com/fecgov/fec-cms/issues/1944 the front-end portion of this ticket?

**Comment by PaulClark2** [comment]
@rjayasekera let me know if you need additional information

A	ADMINISTRATIVELY TERMINATED
M	MONTHLY FILER
Q	QUARTERLY FILER
T	TERMINATED

**Comment by lbeaufort** [comment]
Resolved by fecgov/openFEC#3137
